### ISSUE TITLE
JS of external modules not loaded

### ISSUE THREAD

**Issue by funkycram** [humhub/app#82]
It seams that JS of external modules not loaded.
Example with the Task module (but the problem occurs on other modules too):

![Screenshot_20230615_074134_com humhub app](https://github.com/humhub/app/assets/23310825/aeb49bcc-21a7-4b69-8519-613a244fa0c9)

**Comment by luke-** [comment]
Could reproduce this. 

@PrimozRatej Any idea?

**Comment by ArchBlood** [comment]
I've tested with the verification module and it works fine, seems pjax loaded related?

**Comment by luke-** [comment]
Possibly it affects "<script load>" tags that are inside an Ajax response?

**Comment by funkycram** [comment]
@PrimozRatej if you can't find what you're looking for, don't hesitate to ask me to test what's not working so that I can clarify the problem.

**Comment by ArchBlood** [comment]
> Possibly it affects "<script load>" tags that are inside an Ajax response? 

From my understanding it's related to the nature of how pjax refresh is done, as of now from all my tests with all previous versions and the present version, the external and inline scripts aren't being loaded almost at all till page refresh, this happens with almost all modules, including modules from both Cuzy.app and HumHub/HumHub Contrib, I've not dived deeper into this current, Green Meteor tries it's best to move away from single page loads as most of our modules, after all, most of our modules use external resources from third parties or disable the use of pjax in general. :thinking:

**Comment by PrimozRatej** [comment]
This is the log I get back from a WebView when loading conversations.

```
Console msg: {message: ERROR - root: Could not initialize module: humhub.modules.firebase FirebaseError: Messaging: This browser doesn't support the API's required to use the firebase SDK. (messaging/unsupported-browser)., messageLevel: 3}

Required a non initialized module: mail.inbox.ConversationList

Required a non initialized module: mail.inbox.ConversationList

Required a non initialized module: mail.ConversationView

%cUnhandled rejection @user-script:11:202:21
n@https://community.humhub.com/static/js/humhub-app.js:1:475300
@https://community.humhub.com/static/js/humhub-app.js:1:448836
@https://community.humhub.com/static/js/humhub-app.js:1:449649
@https://community.humhub.com/static/js/humhub-app.js:1:450337
@https://community.humhub.com/static/js/humhub-app.js:1:451668
s@https://community.humhub.com/static/js/humhub-app.js:1:403542
@https://community.humhub.com/static/js/humhub-app.js:1:404659
@https://community.humhub.com/static/js/humhub-app.js:1:403391 color: red
```

The first one I assume is this [firebase-messaging-is-not-supported](https://stackoverflow.com/questions/56215286/firebase-messaging-is-not-supported-in-your-browser-how-to-solve-this)

For the `Unhandled rejection @user-script:`,  no idea. I'm looking into it, but I can't find any good explanation for it.

**Comment by luke-** [comment]
@PrimozRatej That's really strange.

I have a strong assumption that script tags/script loads which are in inside a AJAX response are not being loaded as well.

Of course, this hits many pages after we avoid full page loads. Especially modules (Messages, Calendar, Poll) are affected here, because core javascripts are in a bundle file and are loaded once during a Full Page Load.

---

I have tested other Flutter WebView implementations and they do not show this behavior. However, I am not sure if the identical version of Flutter WebView is used here.

This is working:
https://play.google.com/store/apps/details?id=com.pichillilorenzo.flutter_browser&hl=de

---

Possibly some used options lead to this strange behavior.
https://github.com/humhub/app/blob/master/lib/pages/web_view.dart#L81-L84

Maybe you could try to comment out some or try to build a very reduced WebApp application and then check at which point the erroneous behavior occurs.

--- 

Let me know if I can help.

**Comment by PrimozRatej** [comment]
Figured it out, It seems that there is a problem with the library itself, the moment I override the `shouldInterceptAjaxRequest` the external js files are not correctly injected. Couldn't figure out why, tho, When I run the hybrid app inside a browser (so mobile app build for web) the injection of scripts works as expected (it still uses the underlying web view even if it is run in the browser). 

We were using the `shouldInterceptAjaxRequest` to inject custom headers into the web app so that the instance knows that it is running inside the mobile app.

**Fix:** 
I removed the `shouldInterceptAjaxRequest` method the scripts are now loaded ok. 
The injection of the headers was moved to JS and it is run every time on full page reload. 
Tested it on Android and iOS and looks good to me now.
